core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+          ```
+
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'MFARegistrationV2' &&  _['score'] == 9 )
+          ```
+      remediation: |
+        ### MS365 Console
+
+        To update via the ms365 portal:
+
+        1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+        2. Click on "Azure Active Directory" once in the "Admin Centers"
+        3. Applications --> Enterprise applications --> Security --> Conditional Access 
+        4. Make sure there is a Policy with following conditions: 
+           * Grant --> Grant access (Require multifactor authentication)
+           * Users --> All users
+           * Cloud apps or actions --> All cloud apps 
+        5. Make sure that the policy is Enabled 
+ 
+    query: |
+      msgraph.beta.security.latestSecureScores.controlScores.one( _['controlName'] == 'MFARegistrationV2' &&  _['score'] == 9)
+  - uid: mondoo-m365-ensure-that-office-365-passwords-are-not-set-to-expire
+    title: Ensure that Office 365 passwords never Expire
+    severity: 80
+    docs:
+      desc: |
+        This control checks to make sure the Office 365 passwords never expires.  Based on the new research from several Organizations, it has been confirmed that forcing users to change their password, make the password less secure!
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.domains { passwordValidityPeriodInDays == 2147483647 }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.domains { passwordValidityPeriodInDays == 2147483647 }
+          ```
+      remediation: |
+       ### MS365 Console
+
+        To update via the ms365 portal:
+
+        1. Log in as Global Administrator to the ms365 portal at https://admin.microsoft.com
+        2. Settings --> Org Settings --> Security & Privacy --> Password expiration policy -->  Set passwords to never expire (recommended)
+        3. Click Save 
+    query: |
+      msgraph.beta.domains {
```

Comment:
`msgraph.beta.domains` is deprecated, it should be `microsoft.domains`

Reply:
Thanks Chris, done!